timemachine/fe/free_energy.py

```diff
@@ -139,30 +139,40 @@ def __post_init__(self):
         assert self.batch_size <= self.n_proposals
 
 
+@dataclass(frozen=True)
+class LocalMDParams:
+    local_steps: int
+    k: float = 1_000.0  # kJ/mol/nm^4
+    min_radius: float = 1.0  # nm
+    max_radius: float = 3.0  # nm
+    freeze_reference: bool = True
+
+    def __post_init__(self):
+        assert 0.1 <= self.min_radius <= self.max_radius
+        assert self.local_steps > 0
+        assert 1.0 <= self.k <= 1.0e6
+
+
 @dataclass(frozen=True)
 class MDParams:
     n_frames: int
     n_eq_steps: int
     steps_per_frame: int
     seed: int
-    local_steps: int = 0
-    k: float = 1_000.0  # kJ/mol/nm^4
-    min_radius: float = 1.0  # nm
-    max_radius: float = 3.0  # nm
-    freeze_reference: bool = True
 
+    # Set to LocalMDParams to enable local MD
+    local_md_params: LocalMDParams | None = None
     # Set to HREXParams or None to disable HREX
-    hrex_params: Optional[HREXParams] = None
+    hrex_params: HREXParams | None = None
```

Comment:
nit: should we adopt a convention for this? Personally I'm OK with replacing `Optional[A]` with `A | None`, but would prefer we do this consistently if at all.

Reply:
OK. Mostly just trying to understand what's essential to the PR vs. flipping conventions.